### DISCUSSION TITLE
Why no way to pass Play Integrity?

### DISCUSSION THREAD

**Discussion by FreedomLabsIO** [LSPosed/MagiskOnWSALocal#801] [Q&A]
```
How to pass Play Integrity (formerly known as SafetyNet)?

Like all the other emulators, no way.
```
I understand if we're talking about emulating a newer device that has a hardware Play Integrity chip. But what about older models that doesn't have this chip. What else stops us from passing it if we emulate all the needed properties, build fingerprint, etc?

**Comment by JasonKhew96** [comment]
If Play Integrity can't detect non-genuine Android device, it shouldn't exists in the first place.
